js/store.js

```diff
@@ -1,19 +1,34 @@
 const store = {
+  _websites: null,
+
   init() {
-    this.addListeners();
+    this._read();
+  },
+
+  async _read() {
+    const data = await browser.storage.local.get('websites');
+    this._websites = clone(data.websites);
+
+    return this._websites;
+  },
+
+  async _write(websites) {
+    this._websites = clone(websites);
+
+    return await browser.storage.local.set({ websites });
   },
 
   async getAll() {
-    return await browser.storage.local.get();
+    return await this._read();
   },
 
   async getFirstParty(hostname) {
     if (!hostname) {
       throw new Error('getFirstParty requires a valid hostname argument');
     }
 
-    const storage = await this.getAll();
-    return storage.websites[hostname];
+    const websites = await this.getAll();
+    return websites[hostname];
   },
 
   async getThirdParties(hostname) {
```

Comment:
@princiya I am working on updating [capture.js](https://github.com/electrolyfish/lightbeam-we/blob/master/js/capture.js) per our discussion a few days ago (DNM PR coming shortly!). `hostname` is actually a key used for all three URLs (`document`, `target`, `origin`) by the webextension APIs I use in `capture.js`, so it may not be the best term when referring to the visited site's URL. What if we used `document` instead?